src/lib/transition_router/initial_validator.ml

```diff
@@ -309,6 +320,16 @@ let run ~logger ~trust_system ~verifier ~transition_reader
                       , time_received ) ;
                     return ()
                 | Error error ->
+                    Internal_tracing.with_state_hash state_hash
+                    @@ fun () ->
+                    [%log internal] "Failure"
```

Comment:
Should this log message be more specific? `Failure` seems very broad.

Reply:
Check the `~metadata` argument, the reason for the failure is attached there.

Reply:
In the GCP stackdriver view, this will show up as the message `Failure`, and you have to manually expand to get the metadata. If there's a more descriptive message that could go here instead, that would help us interpret the tracing messages while looking at the raw logs.

Reply:
I see. Let me think about it for a bit, but I think something like `Failure:<arbitrary output>` (with `:` or `\t` or some specific separator I can use to ignore the remaining text in the consumer) would do. Would that be good enough? (I don't know how it looks like in stackdriver).

Reply:
@mrmr1993 btw just to make sure we are on the same page. The files produced by the internal tracing don't look like the JSON logs produced by the node, they just contain a sequence of events and control commands that look like this:

```jsonl
{"current_block":"3NL3n4UnfDEmbx2hap27Z3z7PU1D4pktAd9qmj2PvnCUAhnNT4rQ"}
["Produce_state_transition_proof",1679321390.6394472]
{"metadata":{"transactions_count":0}}
{"current_block":"3NLqqdkAQnbNorXKRFuWC9fz45jvdF1XUiwtseWzRbCB8vaxUpkS"}
{"block_metadata":{"blockchain_length":"2"}}
["External_block_received",1679321408.8380573]
["Initial_validation",1679321408.8467133]
["Verify_blockchain_snarks",1679321408.8499875]
{"metadata":{"count":1}}
{"current_block":"3NL3n4UnfDEmbx2hap27Z3z7PU1D4pktAd9qmj2PvnCUAhnNT4rQ"}
["Produce_chain_transition_proof",1679321409.389129]
["Produce_validated_transition",1679321409.3892083]
["Build_breadcrumb",1679321409.3894844]
{"block_metadata":{"coinbase_receiver":"B62qnzLUoQaPPM33X6JpVJUMq36QSibEgZep4U2idqFwxj3GXK8drXF","creator":"B62qnzLUoQaPPM33X6JpVJUMq36QSibEgZep4U2idqFwxj3GXK8drXF","global_slot":"3","previous_state_hash":"3NLSc2Y4MkCgPHGbfeuNX23Hpk5ZPs9LHgegf28DcMWWVKCxz9wo","slot":"3","winner":"B62qogjiboJ1tsGm42AtnWuxiL6yX3qDb2CGFVtTgD9iEjxGSEQ3uzm"}}
["Validate_staged_ledger_diff",1679321409.3895411]
["Prediff",1679321409.3908107]
["Verify_commands",1679321409.390832]
{"metadata":{"count":0}}
["Verify_commands_done",1679321415.2565856]
["Apply_diff",1679321415.2566586]
["Update_coinbase_stack",1679321415.2567432]
{"metadata":{"coinbases":1,"commands_count":0,"max_throughput":128,"proofs_waiting":0,"spots_available":128,"transactions":1,"works":0}}
["Update_ledger_and_get_statements",1679321415.2567675]
{"metadata":{"partition":"single"}}
["Update_ledger_and_get_statements_done",1679321415.2621703]
["Update_coinbase_stack_done",1679321415.2622075]
{"metadata":{"data_len":1,"is_new_stack":true,"transactions_len":1}}
["Check_for_sufficient_snark_work",1679321415.2622213]
{"metadata":{"free_space":128,"required_pairs":0,"slots":1,"work_count":0}}
["Check_zero_fee_excess",1679321415.262237]
["Fill_work_and_enqueue_transactions",1679321415.262251]
{"metadata":{"emitted_proof":false,"merge_jobs_created":0,"scan_state_added_works":0,"total_proofs":0}}
["Update_pending_coinbase_collection",1679321415.262411]
```

I mention this because I don't know what is the format expected by stackdriver. Obviously, if needed another log processor+transport can be registered that outputs lots in the same format as the rest of the logs.

In any case, if it is useful I can add extra text after the failure (something like a comment so it looks like `["Failure // Validation callback expired", 1679321415.43431]`)